lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -875,7 +956,7 @@ public ObjectTypeParsingRules getTypeDescriptor(final SearchObjectType type)
             }
             if (guidsToLoad.size() > 0) {
                 final List<ObjectData> data =
-                        retrier.retryFunc(g -> getObjectsByIds(g), guidsToLoad, null);
+                        retrier.retryFunc(g -> getObjectsByIds(objectRefPath, g), guidsToLoad, null);
```

Comment:
This seems like a really expensive way to get the type of the object since it now fetches the object from the workspace just to get a type, which can be fetched much more cheaply.

Reply:
Optimized type fetching in MOPLookupProvider